### PR TITLE
Added ability to rename variables far from cursor

### PR DESCRIPTION
**Pull request by PaoloMonti42**
I added the possibility to rename a variable in Visual mode even without the need of selecting it through the cursor (which is what `dn` does already).

For example, in order to change `local_378h`:
```
[0x00402a02 8% 290 /bin/ls]> pd $r @ main+2 # 0x402a02                                                                                                                     
|           0x00402a02      4156           push r14                                                                                                                        
|           0x00402a04      4155           push r13                                                                                                                        
|           0x00402a06      4154           push r12                                                                                                                        
|           0x00402a08      55             push rbp                                                                                                                        
|           0x00402a09      53             push rbx                                                                                                                        
|           0x00402a0a      89fb           mov ebx, edi                                                                                                                    
|           0x00402a0c      4889f5         mov rbp, rsi                                                                                                                    
|           0x00402a0f      4881ec880300.  sub rsp, 0x388                                                                                                                  
|           0x00402a16      488b3e         mov rdi, qword [rsi]                                                                                                            
|           0x00402a19      64488b042528.  mov rax, qword fs:[0x28]    ; [0x28:8]=-1 ; '(' ; 40                                                                            
|           0x00402a22      488984247803.  mov qword [local_378h], rax                                                                                                     
|           0x00402a2a      31c0           xor eax, eax                                                                                                                    
|           0x00402a2c      e8cfb00000     call sub.strrchr_b00        ;[1] ; char *strrchr(const char *s, int c)                                                          
|           0x00402a31      bec19a4100     mov esi, 0x419ac1                                                                                                               
|           0x00402a36      bf06000000     mov edi, 6      
```
The user can type `dv`, then specify '22'.  The function will look for the first instruction containing a variable whose address ends in '22'. If such an instruction is found, the variable will be renamed.

### CODE REVIEW

File: libr/core/vmenus.c [hunk 1]
```diff
@@ -153,7 +153,7 @@ R_API bool r_core_visual_esil(RCore *core) {
 		// bool use_color = core->print->flags & R_PRINT_FLAGS_COLOR;
 		(void) r_asm_disassemble (core->assembler, &asmop, buf, sizeof (ut64));
 		analop.type = -1;
-		(void)r_anal_op (core->anal, &analop, core->offset, buf, sizeof (ut64), R_ANAL_OP_MASK_ESIL);
+		(void)r_anal_op (core->anal, &analop, core->offset, buf, sizeof (ut64), R_ANAL_OP_MASK_ALL);
```
Comment (sivaramaaa) [inline libr/core/vmenus.c]: Why do u need R_ANAL_OP_MASK_ALL , where only MASK_ESIL is needed ?

File: libr/core/vmenus.c [hunk 2]
```diff
@@ -292,7 +292,7 @@ static bool edit_bits (RCore *core) {
 		bool use_color = core->print->flags & R_PRINT_FLAGS_COLOR;
 		(void) r_asm_disassemble (core->assembler, &asmop, buf, sizeof (ut64));
 		analop.type = -1;
-		(void)r_anal_op (core->anal, &analop, core->offset, buf, sizeof (ut64), R_ANAL_OP_MASK_ESIL);
+		(void)r_anal_op (core->anal, &analop, core->offset, buf, sizeof (ut64), R_ANAL_OP_MASK_ALL);
```
Comment (sivaramaaa) [inline libr/core/vmenus.c]: same goes here and other few places !

File: libr/core/vmenus.c [hunk 3]
```diff
@@ -3130,6 +3133,81 @@ R_API void r_core_visual_define (RCore *core, const char *args) {
 			}
 		}
 		break;
+        case 'v':
+        {
+		RAnalOp op;
+
```
Comment (sivaramaaa) [inline libr/core/vmenus.c]: No need of newline

File: libr/core/vmenus.c [hunk 4]
```diff
@@ -3130,6 +3133,81 @@ R_API void r_core_visual_define (RCore *core, const char *args) {
...
+		ut64 N;
+		char *end_off = r_cons_input ("Last hexadecimal digits of instruction: ");
+		int ret = sscanf (end_off,"%lx\n", &N);
+		if (!ret) {
+			eprintf ("Invalid numeric input\n");
+			r_sys_sleep (1);
+			break;
+		}
+                free (end_off);
```
Comment (sivaramaaa) [inline libr/core/vmenus.c]: use tabs and not spaces

File: libr/core/vmenus.c [hunk 5]
```diff
@@ -3130,6 +3133,81 @@ R_API void r_core_visual_define (RCore *core, const char *args) {
...
+
+		ut64 incr = 0x10;
+                ut64 tmp_N = N >> 4;
```
Comment (sivaramaaa) [inline libr/core/vmenus.c]: same here tabs

File: libr/core/vmenus.c (hunk 1 above)
Comment (PaoloMonti42) [reply libr/core/vmenus.c]: To be fair, I have absolutely no idea why that line changed.
I didn't even acknowledge that code, I only wrote mine and fixed spaces around.

Comment (sivaramaaa) [reply libr/core/vmenus.c]: Ah np :) , can u try to rebase this on top of master, may solve the problem

File: libr/core/vmenus.c [hunk 6]
```diff
@@ -2599,19 +2599,21 @@ R_API void r_core_seek_next(RCore *core, const char *type) {
 	ut64 next = UT64_MAX;
 	if (strstr (type, "opc")) {
 		RAnalOp aop;
-		if (r_anal_op (core->anal, &aop, core->offset, core->block, core->blocksize, R_ANAL_OP_MASK_BASIC)) {
```
Comment (sivaramaaa) [inline libr/core/vmenus.c]: One more u missed it :)

Comment (PaoloMonti42) [reply libr/core/vmenus.c]: Nothing better to do on saturday morning than look at my fuckups, huh?

Comment (sivaramaaa) [reply libr/core/vmenus.c]: It happens to everyone  np :) , the rest will be reviewd by @radare

File: libr/core/vmenus.c [hunk 7]
```diff
@@ -3130,6 +3133,80 @@ R_API void r_core_visual_define (RCore *core, const char *args) {
 			}
 		}
 		break;
+        case 'v':
+        {
+		RAnalOp op;
+		ut64 N;
+		char *end_off = r_cons_input ("Last hexadecimal digits of instruction: ");
+		int ret = sscanf (end_off,"%lx\n", &N);
```
Comment (radare) [inline libr/core/vmenus.c]: better use `ut64 N = strtoull(end_off, NULL, 16);`

File: libr/core/vmenus.c [hunk 8]
```diff
@@ -1702,7 +1702,7 @@ R_API void r_core_visual_config(RCore *core) {
 				if (show) {
 					if (option == i) hit = 1;
 					if ( (i >=option-delta) && ((i<option+delta)||((option<delta)&&(i<(delta<<1))))) {
-						r_cons_printf(" %c  %s\n", (option==i)?'>':' ', old);
+						r_cons_printf (" %c  %s\n", (option==i)?'>':' ', old);
```
Comment (radare) [inline libr/core/vmenus.c]: spaces around ==

**Pull request closed without merging**